projects/packages/videopress/src/client/components/video-upload-area/index.tsx

```diff
@@ -0,0 +1,103 @@
+import { Button, useBreakpointMatch, Text } from '@automattic/jetpack-components';
+import { Spinner } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { Icon, cloudUpload } from '@wordpress/icons';
+import classnames from 'classnames';
+import { DragEvent, useCallback, useState } from 'react';
+import styles from './style.module.scss';
+import { VideoUploadAreaProps } from './types';
+import type React from 'react';
+
+/**
+ * Video Upload Area component
+ *
+ * @param {VideoUploadAreaProps} props - Component props.
+ * @returns {React.ReactNode} - VideoUploadArea react component.
+ */
+const VideoUploadArea: React.FC< VideoUploadAreaProps > = ( {
+	isLoading = false,
+	className,
+	onSelectFile,
+} ) => {
+	const [ isSm ] = useBreakpointMatch( 'sm' );
+	const [ isDraggingOver, setIsDraggingOver ] = useState( false );
+
+	const handleClickEvent = useCallback( () => {
+		const input = document.createElement( 'input' );
+		input.type = 'file';
+
+		input.addEventListener( 'change', ( e: Event ) => {
+			const target = e.target as HTMLInputElement;
+			onSelectFile( target.files[ 0 ] );
+			input.remove();
+		} );
+
+		input.click();
+	}, [ onSelectFile ] );
+
+	const handleDragOverEvent = useCallback( ( event: DragEvent< HTMLDivElement > ) => {
+		event.preventDefault();
+		setIsDraggingOver( true );
+	}, [] );
+
+	const handleDragLeaveEvent = useCallback( () => {
+		setIsDraggingOver( false );
+	}, [] );
+
+	const handleDropEvent = useCallback(
+		( event: DragEvent< HTMLDivElement > ) => {
+			event.preventDefault();
+
+			const { files } = event.dataTransfer;
+			if ( files.length > 1 ) {
+				throw new Error( __( 'Only one file allowed', 'jetpack-videopress-pkg' ) );
+			}
+
+			onSelectFile( files[ 0 ] );
+			setIsDraggingOver( false );
+		},
+		[ onSelectFile ]
+	);
+
+	return isLoading ? (
```

Comment:
I suggest keeping the wrapper and changing the content, already adding a base `height` for the wrapper.

This will avoid the height change from upload to loading, which blinks the UI.